### PR TITLE
feat(dialog): add CSS variables to offset the position

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #9879

## Summary

- `--calcite-dialog-offset-x`: Specifies the horizontal offset of the component.
- `--calcite-dialog-offset-y`: Specifies the vertical offset of the component.

### CODE REVIEW

File: packages/calcite-components/src/components/dialog/dialog.scss [hunk 1]
```diff
@@ -9,6 +9,8 @@
  * @prop --calcite-dialog-content-space: Specifies the padding of the component's content.
  * @prop --calcite-dialog-footer-space: Specifies the padding of the component's footer.
  * @prop --calcite-dialog-border-color: Specifies the component's border color.
+ * @prop --calcite-dialog-offset-x: Specifies the horizontal offset of the component.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/dialog/dialog.scss:12]: `offset` makes sense to me!

**Review by macandcheese — APPROVED** [review]
Nice!

**Review by jcfranco — APPROVED** [review]
👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍
👍🤓👍👍👍🤓🤓🤓🤓👍🤓🤓🤓🤓🤓👍🤓👍👍👍🤓👍🤓👍
👍🤓👍👍👍🤓👍👍👍👍👍👍🤓👍👍👍🤓🤓👍🤓🤓👍🤓👍
👍🤓👍👍👍🤓👍🤓🤓👍👍👍🤓👍👍👍🤓👍🤓👍🤓👍🤓👍
👍🤓👍👍👍🤓👍👍🤓👍👍👍🤓👍👍👍🤓👍👍👍🤓👍👍👍
👍🤓🤓🤓👍🤓🤓🤓🤓👍👍👍🤓👍👍👍🤓👍👍👍🤓👍🤓👍
👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍
 
I _think_ we might be able to leverage out-of-the-box CSS props for this if we make adjustments similar to [this](https://github.com/Esri/calcite-design-system/pull/9545). I'll explore this a bit and will create a refactor issue if it ends up working.

**Comment by jcfranco** [comment]
@alisonailea Can you take a quick look at the new CSS props? 🙇‍♂️

**Review by alisonailea — APPROVED** [review]
Looks good!

**Pull request merged**